be/src/util/file_utils.cpp

```diff
@@ -202,12 +219,15 @@ Status FileUtils::split_pathes(const char* path, std::vector<std::string>* path_
     return Status::OK();
 }
 
-Status FileUtils::copy_file(const std::string& src_path, const std::string& dest_path) {
-    ASSIGN_OR_RETURN(auto src_file, Env::Default()->new_sequential_file(src_path));
-
-    ASSIGN_OR_RETURN(auto dest_file, Env::Default()->new_writable_file(dest_path));
-
-    return copy(src_file.get(), dest_file.get()).status();
+Status FileUtils::copy_file(const std::string& src_path, const std::string& dst_path) {
+    ASSIGN_OR_RETURN(auto src_env, Env::CreateSharedFromString(src_path));
+    ASSIGN_OR_RETURN(auto dst_env, Env::CreateSharedFromString(dst_path));
+    ASSIGN_OR_RETURN(auto src_file, src_env->new_sequential_file(src_path));
+    ASSIGN_OR_RETURN(auto dst_file, dst_env->new_writable_file(dst_path));
+    RETURN_IF_ERROR(copy(src_file.get(), dst_file.get()));
+    RETURN_IF_ERROR(dst_file->sync());
+    RETURN_IF_ERROR(dst_file->close());
```

Comment:
close is not needed.

Reply:
> close is not needed.

Destruct a WritableFile without close() is undefined behavior, for example, the WritableFile for S3 will not flush data if close() not been called.